go-controller/pkg/cni/cni.go

```diff
@@ -146,22 +172,38 @@ func (pr *PodRequest) cmdAddWithGetCNIResultFunc(
 			if err = pr.addDPUConnectionDetailsAnnot(kubecli, clientset.podLister, netdevName); err != nil {
 				return nil, err
 			}
-			annotCondFn = isDPUReady
+			// defer DPU connection details check so primary UDN annotation check can be done in parallel if any
 		}
 		// In the case of SmartNIC (CX5), we store the netdevname in the representor's
 		// OVS interface's external_id column. This is done in ConfigureInterface().
 	}
 	// Get the IP address and MAC address of the pod
 	// for DPU, ensure connection-details is present
 
-	primaryUDN := udn.NewPrimaryNetwork(networkManager)
+	primaryUDN := udn.NewPrimaryNetwork(networkManager, clientset.nadLister)
 	if util.IsNetworkSegmentationSupportEnabled() {
-		annotCondFn = primaryUDN.WaitForPrimaryAnnotationFn(podName, namespace, annotCondFn)
+		annotCondFn = primaryUDN.WaitForPrimaryAnnotationFn(annotCondFn)
+		annotCondFn = pr.primaryDPUReady(primaryUDN, kubecli, clientset.podLister, annotCondFn)
+	}
+
+	// now checks DPU connection details
+	if config.OvnKubeNode.Mode == types.NodeModeDPUHost {
+		if pr.CNIConf.DeviceID != "" {
+			annotCondFn = isDPUReady(annotCondFn, pr.nadName)
```

Comment:
isn't isDPUReady going to be called twice for UDNs on DPU host? primaryDPUReady will call it then it will get called again in this? Or is this for default network?

Reply:
yes, primaryDPUReady is called to check primary udn network and this is one is for default network. let me add comments to make it clear